ec-europa/eubfr-data-lake
# feat(facade): refactoring - EUBFR-14

- facade service moves 1 level up out of ingestion group
- `serverless-offline` takes care of resources now
- it also takes care for local development and routing to handlers
- CF stack still retains full permissions in order to be able to just update the stack
- API key validation still holds true after deployment, it's off when local dev on private methods
- swagger spec is updated

Keep in mind that staging variable acts differently for `serverless-offline`. This means that your config
```json
{
  "region": "eu-central-1",
  "stage": "devkc"
}
```
The `stage` variable should be simpler chan `dev-{username}-{random}` as this is used for staging endpoint of API gateway. If sophisticated variable, this error occurs on CF provisioning:
```
An error occurred: ApiGatewayDeployment1506343257753 - Stage name only allows a-zA-Z0-9_
```

To test the pull request:
- update your `config.json`
- run `yarn` to get new dependencies
- run `yarn deploy` within the service and see the CF creates or updates
- follow the information from the output of CF to test your api
- if you create a new API, add it to the usage plan for API keys in the API Gateway console

Currently, the functions as logging event and context information to CloudWatch, you might want to check these to see information about user and request.
